src/Raven.Server/Documents/Indexes/Static/Roslyn/Rewriters/ReduceIndex/GroupByFieldsRetriever.cs

```diff
@@ -146,20 +146,45 @@ public override SyntaxNode VisitGroupClause(GroupClauseSyntax node)
 
                 by = by.Replace($"{result}.", string.Empty);
 
-                GroupByFields = by.Split(',');
+                var fields = by.Split(',');
 
-                for (int i = 0; i < GroupByFields.Length; i++)
+                var groupByFields = new List<string>(fields.Length);
+
+                for (var i = 0; i < fields.Length; i++)
                 {
-                    var field = GroupByFields[i];
+                    var parts = fields[i].Split('=');
 
-                    var parts = field.Split('=');
+                    string fieldName;
 
                     if (parts.Length == 1)
-                        GroupByFields[i] = parts[0].Trim();
+                        fieldName = parts[0].Trim();
                     else
-                        GroupByFields[i] = parts[1].Trim();
+                        fieldName = parts[1].Trim();
+
+                    if (fieldName.StartsWith('"') && fieldName.EndsWith('"'))
```

Comment:
And if it is quoted by `'` ?

Reply:
It's for static indexes only. It won't be quoted.